### PR TITLE
Update .mailmap

### PR DESCRIPTION
**Pull request by SooluThomas**
### Summary

Updating mailmap to remove my IBM email ID. This is to check if the GitHub API for generating the contributors list would list my account anonymous or not.

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -47,7 +47,7 @@ Salvador de la Puente González <neo.salvador@gmail.com> <salva@unoyunodiez.com>
 Shaohan Hu <shaohan.hu@ibm.com> <hushaohan@gmail.com>
 Shelly Garion <46566946+ShellyGarion@users.noreply.github.com>
 Simone Perriello <simone.perriello@protonmail.com>
-Soolu Thomas <soolu.thomas@ibm.com> <soolu.elto@gmail.com>
+Soolu Thomas <soolu.elto@gmail.com>
```
Comment (kevinhartman) [inline .mailmap]: I think what you might want here is to just swap these if you're committing from `<soolu.thomas@ibm.com>` and want your canonical email address to be `<soolu.elto@gmail.com>`?

Comment (SooluThomas) [reply .mailmap]: Yeah, that sounds fair to me. But I don't really commit form the ibm email. But I definitely can make that change

Comment (SooluThomas) [reply .mailmap]: Fixed in https://github.com/Qiskit/qiskit/pull/1656/commits/00bc20c9b5b29739e91eabef4bcf3046e311dafd

**Review by kevinhartman — APPROVED** [review]
LGTM

**Pull request merged**